### ISSUE TITLE
nix-repl: segmentation fault

### ISSUE THREAD

**Issue by danbst** [NixOS/nix#860]
I'm on 15.09-small, and nix-repl doesn't work, just segfaults.

Here are some details
```
# readlink $(which nix-repl)
/nix/store/xvi58q959p389d44lqibwdi7fpnq3c9i-nix-repl-1.10-8a2f5f0/bin/nix-repl

# gdb nix-repl
....
(gdb) run
...
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/nix/store/7b82878b8l4d136i6pxmwcc12sfsda33-glibc-2.21/lib/libthread_db.so.1".
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7734501 in nix::downloadFileCached(std::string const&, bool) () from /nix/store/nh2kih74f47fhnbf2bg5q49vi2f9jq8l-nix-1.10/lib/libnixstore.so
```

**Comment by joachifm** [comment]
I had the same problem, deleting `~/.cache/nix/tarballs` fixed it for me ... never figured out what the problem actually was. The only thing of note was that the tarballs cache folder contained a few dangling links that I couldn't delete directly for whatever reason.

**Comment by danbst** [comment]
Right, 
```
# ls -la .cache/nix/tarballs/
total 12
drwxr-xr-x 2 root root 4096 Mar 24 15:11 .
drwxr-xr-x 3 root root 4096 Mar 23 11:54 ..
lrwxrwxrwx 1 root root   59 Mar 24 14:28 0y32f2w0byynlq1147x6v8f3awck39qdhdcc6pzfiz25dadlg0aj-file -> /nix/store/v6191syx6zqfkdbcg1bdanz3lhkhwncv-nixexprs.tar.xz
-rw-r--r-- 1 root root  103 Mar 24 14:28 0y32f2w0byynlq1147x6v8f3awck39qdhdcc6pzfiz25dadlg0aj.info
lrwxrwxrwx 1 root root   59 Mar 24 15:11 v6191syx6zqfkdbcg1bdanz3lhkhwncv-nixexprs.tar.xz-unpacked -> /nix/store/f730vrl65va66hhjfi0g5irafi37v43s-nixexprs.tar.xz

# ls /nix/store/v6191syx6zqfkdbcg1bdanz3lhkhwncv-nixexprs.tar.xz
ls: cannot access /nix/store/v6191syx6zqfkdbcg1bdanz3lhkhwncv-nixexprs.tar.xz: No such file or directory

# ls /nix/store/f730vrl65va66hhjfi0g5irafi37v43s-nixexprs.tar.xz
ls: cannot access /nix/store/f730vrl65va66hhjfi0g5irafi37v43s-nixexprs.tar.xz: No such file or directory
```

Though removing them didn't help
```
# nix-repl
downloading ‘https://nixos.org/channels/nixos-15.09-small/nixexprs.tar.xz’...
Segmentation fault
```
I have this nix path: `nixpkgs=https://nixos.org/channels/nixos-15.09-small/nixexprs.tar.xz:nixos-config=/etc/nixos/configuration.nix`

**Comment by joachifm** [comment]
Sorry, the tarballs folder was a red herring, I forgot that I had changed the nix path as well. I can reproduce this problem consistently. Starting up nix-repl consumes all my memory and crashes, when the nixpkgs points to a nixexprs tarball, like the one used by @danbst. The problem does not occur if nixpkgs points to an unpacked collection of nixexprs.

**Comment by rimmington** [comment]
Looks like this was reported upstream: https://github.com/edolstra/nix-repl/issues/31

**Comment by domenkozar** [comment]
Let's close this as it's an upstream bug.

**Comment by domenkozar** [comment]
FYI: this is now fixed on master and 16.03, wait for channel updates. See https://github.com/edolstra/nix-repl/issues/31
